src/components/data/data-utils.js

```diff
@@ -472,68 +472,217 @@ export const EntityRenamer = ({ entityType, entityName, workspaceId: { namespace
-  ))
-  const [isBusy, setIsBusy] = useState()
-  const [consideringDelete, setConsideringDelete] = useState()
+  return { type, isList }
+}
 
-  const isList = _.isArray(newValue)
-  const isUnchanged = (initialType === editType && initialIsList === isList && _.isEqual(attributeValue, newValue))
+export const convertAttributeValue = (attributeValue, newType, referenceEntityType) => {
+  if (newType === 'reference' && !referenceEntityType) {
+    throw new Error('An entity type is required to convert an attribute to a reference')
+  }
 
-  const makeTextInput = placeholder => ({ value = '', ...props }) => h(TextInput, { autoFocus: true, placeholder, value, ...props })
+  const { type, isList } = getAttributeType(attributeValue)
 
-  const { prepForUpload, makeInput, blankVal } = Utils.switchCase(editType,
-    ['string', () => ({
-      prepForUpload: _.trim,
-      makeInput: makeTextInput('Enter a value'),
-      blankVal: ''
-    })],
-    ['reference', () => ({
-      prepForUpload: v => ({ entityName: _.trim(v), entityType: linkedEntityType }),
-      makeInput: makeTextInput(`Enter a ${linkedEntityType}_id`),
-      blankVal: ''
-    })],
-    ['number', () => ({
-      prepForUpload: _.identity,
-      makeInput: ({ value = 0, ...props }) => h(NumberInput, { autoFocus: true, isClearable: false, value, ...props }),
-      blankVal: 0
+  const baseConvertFn = Utils.switchCase(
+    newType,
+    ['reference', () => value => ({
+      entityType: referenceEntityType,
+      entityName: _.toString(value) // eslint-disable-line lodash-fp/preferred-alias
     })],
-    ['boolean', () => ({
-      prepForUpload: _.identity,
-      makeInput: ({ value = false, ...props }) => div({ style: { flexGrow: 1, display: 'flex', alignItems: 'center', height: '2.25rem' } },
-        [h(Switch, { checked: value, ...props })]),
-      blankVal: false
-    })]
+    ['number', () => value => {
+      const numberVal = _.toNumber(value)
+      return _.isNaN(numberVal) ? 0 : numberVal
+    }],
+    [Utils.DEFAULT, () => Utils.convertValue(newType)]
   )
+  const convertFn = type === 'reference' ? _.flow(_.get('entityName'), baseConvertFn) : baseConvertFn
+
+  if (isList) {
+    return _.flow(
+      _.update('items', _.map(convertFn)),
+      newType === 'reference' ? _.set('itemsType', 'EntityReference') : _.omit('itemsType')
+    )(attributeValue)
+  }
+
+  return convertFn(attributeValue)
+}
+
+const renderInputForAttributeType = _.curry((attributeType, props) => {
+  return Utils.switchCase(attributeType,
+    ['string', () => {
+      const { value = '', onChange, ...otherProps } = props
+      return h(TextInput, {
+        autoFocus: true,
+        placeholder: 'Enter a value',
+        value,
+        onChange: v => onChange(_.trim(v)),
+        ...otherProps
+      })
+    }],
+    ['reference', () => {
+      const { value, onChange, ...otherProps } = props
+      return h(TextInput, {
+        autoFocus: true,
+        placeholder: 'Enter a value',
```

Comment:
This used to be `Enter a ${linkedEntityType}_id`. Was the change intentional?

Reply:
Oops, that was not intentional.

Reply:
Fixed in [5342971](https://github.com/DataBiosphere/terra-ui/pull/2895/commits/5342971e15085e1c826f60e9c87852c1c551f3df).